BUG: Test flake in `--setup` tests.

The new `--setup` tests exec real processes and may be a bit to wonky to keep in their present form. Anecdotally this is happening frequently enough that we should investigate more.

Failed CI example: https://travis-ci.org/FormidableLabs/builder/jobs/107229385

### runs with --setup

```
 1) bin/builder-core builder run runs with --setup:
     Uncaught AssertionError: expected write to have been called with arguments matching SETUP
    write(1 - BAR_TASK
) at Socket.ondata (_stream_readable.js:540:20)
    write(2 - BAR_TASK
) at Socket.ondata (_stream_readable.js:540:20)
    write(3 - BAR_TASK
) at Socket.ondata (_stream_readable.js:540:20)
    write(4 - BAR_TASK
) at Socket.ondata (_stream_readable.js:540:20)
    write(5 - BAR_TASK
) at Socket.ondata (_stream_readable.js:540:20)
    write(6 - BAR_TASK
) at Socket.ondata (_stream_readable.js:540:20)
    write(EXIT - BAR_TASK - 0) at Socket.ondata (_stream_readable.js:540:20)
      at test/server/spec/bin/builder-core.spec.js:279:18
      at bin/builder-core.js:9:1893
      at lib/runner.js:9:8828
      at node_modules/async/lib/async.js:52:16
      at node_modules/async/lib/async.js:361:13
      at node_modules/async/lib/async.js:52:16
      at done (node_modules/async/lib/async.js:246:17)
      at node_modules/async/lib/async.js:44:16
      at node_modules/async/lib/async.js:358:17
      at lib/utils/tracker.js:9:1675
      at killAll (node_modules/tree-kill/index.js:62:16)
      at node_modules/tree-kill/index.js:33:13
      at ChildProcess.onClose (node_modules/tree-kill/index.js:89:17)
      at maybeClose (child_process.js:1016:16)
      at Process.ChildProcess._handle.onexit (child_process.js:1088:5)
```

### handles --setup early 0 exit

```
  1) bin/builder-core builder run handles --setup early 0 exit:
     Uncaught TypeError: Cannot read property 'message' of undefined
      at Assertion.<anonymous> (node_modules/chai/lib/chai/core/assertions.js:890:14)
      at Assertion.ctx.(anonymous function) [as property] (node_modules/chai/lib/chai/utils/addMethod.js:41:25)
      at test/server/spec/bin/builder-core.spec.js:303:29
      at bin/builder-core.js:9:1893
      at lib/runner.js:9:8828
      at node_modules/async/lib/async.js:52:16
      at node_modules/async/lib/async.js:361:13
      at node_modules/async/lib/async.js:52:16
      at done (node_modules/async/lib/async.js:246:17)
      at node_modules/async/lib/async.js:44:16
      at node_modules/async/lib/async.js:358:17
      at lib/utils/tracker.js:9:1675
      at killAll (node_modules/tree-kill/index.js:62:16)
      at node_modules/tree-kill/index.js:33:13
      at ChildProcess.onClose (node_modules/tree-kill/index.js:89:17)
      at maybeClose (child_process.js:1016:16)
      at Socket.<anonymous> (child_process.js:1184:11)
      at Pipe.close (net.js:484:11)
```

--- a/test/server/spec/bin/builder-core.spec.js
+++ b/test/server/spec/bin/builder-core.spec.js
@@ -467,7 +467,7 @@ describe("bin/builder-core", function () {
         "package.json": JSON.stringify({
           "scripts": {
             "setup": "node test/server/fixtures/repeat-script.js 2 SETUP",
-            "bar": "node test/server/fixtures/repeat-script.js 5 BAR_TASK"
+            "bar": "node test/server/fixtures/repeat-script.js 20 BAR_TASK"
           }
         }, null, 2)
       });
@@ -494,7 +494,7 @@ describe("bin/builder-core", function () {
         "package.json": JSON.stringify({
           "scripts": {
             "setup": "node test/server/fixtures/repeat-script.js 2 SETUP 1",
-            "bar": "node test/server/fixtures/repeat-script.js 10 BAR_TASK"
+            "bar": "node test/server/fixtures/repeat-script.js 20 BAR_TASK"
           }
         }, null, 2)
       });

--- a/test/server/spec/bin/builder-core.spec.js
+++ b/test/server/spec/bin/builder-core.spec.js
@@ -432,9 +432,7 @@ describe("bin/builder-core", function () {
     // https://github.com/FormidableLabs/builder/issues/9
     it("overrides a <archetype> command with a <root> one in a composed <archetype> command");
 
-    // TODO: Fix flake in --setup tests.
-    // https://github.com/FormidableLabs/builder/issues/86
-    it.skip("runs with --setup", function (done) {
+    it("runs with --setup", function (done) {
       base.sandbox.spy(Task.prototype, "run");
       base.mockFs({
         "package.json": JSON.stringify({
@@ -463,9 +461,7 @@ describe("bin/builder-core", function () {
 
     });
 
-    // TODO: Fix flake in --setup tests.
-    // https://github.com/FormidableLabs/builder/issues/86
-    it.skip("handles --setup early 0 exit", function (done) {
+    it("handles --setup early 0 exit", function (done) {
       base.sandbox.spy(Task.prototype, "run");
       base.mockFs({
         "package.json": JSON.stringify({
@@ -492,9 +488,7 @@ describe("bin/builder-core", function () {
 
     });
 
-    // TODO: Fix flake in --setup tests.
-    // https://github.com/FormidableLabs/builder/issues/86
-    it.skip("handles --setup early 1 exit", function (done) {
+    it("handles --setup early 1 exit", function (done) {
       base.sandbox.spy(Task.prototype, "run");
       base.mockFs({
         "package.json": JSON.stringify({

--- a/test/server/spec/bin/builder-core.spec.js
+++ b/test/server/spec/bin/builder-core.spec.js
@@ -467,7 +467,7 @@ describe("bin/builder-core", function () {
         "package.json": JSON.stringify({
           "scripts": {
             "setup": "node test/server/fixtures/repeat-script.js 2 SETUP",
-            "bar": "node test/server/fixtures/repeat-script.js 20 BAR_TASK"
+            "bar": "sleep 10"
           }
         }, null, 2)
       });
@@ -494,7 +494,7 @@ describe("bin/builder-core", function () {
         "package.json": JSON.stringify({
           "scripts": {
             "setup": "node test/server/fixtures/repeat-script.js 2 SETUP 1",
-            "bar": "node test/server/fixtures/repeat-script.js 20 BAR_TASK"
+            "bar": "sleep 10"
           }
         }, null, 2)
       });

